README
More to come :)


ChangeLog
2008-01-11 06:33  hephooey

	* : update the default address book

2008-01-10 06:32  hephooey

	* : timer to reduce blink when resizing

2008-01-10 05:29  hephooey

	* : show the cursor after refresh screen

2008-01-10 05:29  hephooey

	* : Fix the menu selector, still slow

2008-01-09 14:38  hephooey

	* : fix compilation

2008-01-09 14:29  hephooey

	* : fix the ugly background

2008-01-09 14:23  hephooey

	* : update cvs.sh to support automake 1.10

2008-01-09 10:43  hephooey

	* : fix scale problem

2008-01-09 10:43  hephooey

	* : fix blink

2008-01-09 10:43  hephooey

	* : missing painter.end()

2008-01-09 10:43  hephooey

	* : Colorful copy patch by luohandsome at newsmth.

2008-01-09 10:42  hephooey

	* : remove the debug command and extra space

2008-01-07 12:41  hephooey

	* : update about dialog

2008-01-07 12:13  hephooey

	* : bump the version

2008-01-07 12:13  hephooey

	* : update the RELEASE file

2008-01-07 12:13  hephooey

	* : block input before ssh session is ok.

2008-01-07 12:12  hephooey

	* : hide the popup window

2008-01-07 12:12  hephooey

	* : distinguish URL spans over many rows
	  provide by Liu Yubao <yubao.liu at gmail.com>

2008-01-07 12:12  hephooey

	* : make isIllChar() more strict
	  provide by Liu Yubao <yubao.liu at gmail.com>

2008-01-07 12:12  hephooey

	* : Update Makefile

2008-01-07 12:11  hephooey

	* : Use the new getput function

2008-01-07 12:11  hephooey

	* : Replace the old getput.h to avoid the license problem.

2008-01-07 11:14  hephooey

	* : attemption to fix the binary problem in svn.

2008-01-07 10:35  hephooey

	* : Double buffer to avoid flicker, this is experimental, but since
	  the code stays in svn for so long may be we can give it a shot.

2008-01-07 10:32  hephooey

	* : Revert unrelated changes introduced by accident

2008-01-07 10:31  hephooey

	* : revert the unrelated change introduced by accident

2007-08-12 08:48  hephooey

	* : Playground for QtScript

2006-07-26 14:46  hephooey

	* : workaround for VC

2006-07-26 14:45  hephooey

	* : update the project file

2006-07-26 14:44  hephooey

	* : bump the version

2006-07-26 14:21  hephooey

	* : Windows compilation part1:
	  Do not include *.moc if config.h is not defined

2006-07-24 10:05  hephooey

	* : autoconf 2.60

2006-07-11 10:04  hephooey

	* : remove ex files

2006-07-10 07:25  hephooey

	* : require qt 3.3.x

2006-07-10 07:25  hephooey

	* : bump the version

2006-07-10 07:24  hephooey

	* : release notes

2006-07-10 07:00  hephooey

	* : run the script

2006-03-15 12:37  hephooey

	* : reverse the last commit, fix the icon bug

2006-03-08 13:50  hephooey

	* : simplify

2006-02-19 08:45  hephooey

	* : test the commit notification system

2006-02-17 05:20  hephooey

	* : RELEASE_NOTES for pre4

2006-02-16 14:06  hephooey

	* : ChangeLog generated by cvs2cl

2006-02-16 14:05  hephooey

	* : pre4 is coming

2006-02-16 07:28  hephooey

	* : Workaround for the timefilter problem, QFileInfo->created()
	  only give a second level accuracy

2006-02-12 15:50  hephooey

	* : ugly hack for a regression of the fix of the menu bug

2006-02-12 11:13  hephooey

	* : speed up compilation:
	  
	  real 1m41.394s
	  user 1m33.998s
	  sys 0m6.116s
	  
	  v.s.
	  
	  real 1m11.976s
	  user 1m7.083s
	  sys 0m3.999s

2006-02-08 23:08  sidos

	* : move filename".part" to filename after finished always

2006-02-05 15:07  hephooey

	* : This should fix the mysterious menu bug: Text disappear when it
	  was selected
	  in a menu list.

2006-02-05 15:02  hephooey

	* : Redundant?

2006-02-02 15:49  hephooey

	* : I'm a nut, maybe this bug have something to do with the weird
	  menu bug.

2006-02-01 21:42  sidos

	* : updateList when show back imageviewer

2006-02-01 05:55  hephooey

	* : avoid crash when port to qt4

2006-01-30 23:55  sidos

	* : . single instance imageviewer
	  . only pics from current session are shown

2005-12-28 14:44  hephooey

	* : redirect QIMEvent from qtermwindow to qtermscreen, temporary
	  workaround for qt with immqt patches (typically fc4), there
	  should be a better solution, and this workaround still has not
	  been tested on qt without immqt patches.

2005-11-28 03:43  hephooey

	* : deleteLater

2005-11-24 19:17  sidos

	* : fix one potential problem when analyzing http response for
	  filename

2005-11-15 14:04  hephooey

	* : hmm, seems there is no way to tell the charset of the filename
	  from the header, any better idea?

2005-11-04 06:08  hephooey

	* : fix the compile error

2005-10-22 14:25  hephooey

	* : now statusbar in qtermwindow.cpp has no use, should be
	  disabled, add download process from qtermhttp.cpp directly thanks
	  to the singleton, clean up unneccessary code.

2005-10-22 14:23  hephooey

	* : remove the clock, welcome the progress bar

2005-10-22 14:22  hephooey

	* : make statusBar a singleton, never use the create method outside
	  of qtermframe.cpp.

2005-10-22 14:20  hephooey

	* : more ignore

2005-10-22 14:11  hephooey

	* : dos2unix

2005-10-04 10:08  hephooey

	* : show the progress

2005-10-04 08:54  hephooey

	* : new progress bar on the status bar

2005-10-04 08:38  hephooey

	* : use osd to display ip location and download information

2005-09-21 10:07  sidos

	* : fixed: scrollbar position always set to left after FullScreen
	  mode(chenniqiu at gmail)

2005-09-19 07:46  sidos

	* : fix bugs reported by nichloas@smth.org

2005-09-01 14:49  hephooey

	* : for 0.4pre3

2005-09-01 12:19  hephooey

	* : for 0.4pre3

2005-09-01 12:11  sidos

	* : code correction

2005-09-01 12:03  sidos

	* : fix problem with input under MacOSX

2005-09-01 11:22  sidos

	* : check path ended with '/' or not

2005-09-01 08:58  sidos

	* : use QPixmap::createHeuristicMask

2005-09-01 08:49  sidos

	* : update qterm.prj to include new files

2005-09-01 04:04  hephooey

	* : CHANGE BEHAVIOR:
	  click the middle button on a url will preview the url instead of
	  paste clipboard.

2005-09-01 04:01  hephooey


qterm.h
#ifndef QTERM_H
#define QTERM_H

namespace QTerm
{
    enum Mode {BBS, EDITOR, VIEWER};
    enum ImagePosition {Center, Tile, Stretch};
    enum DisplayOrientation {DisplayHorizontal, DisplayVertical};
    enum CharFlag {Latin1,Cn_First,Cn_Second};
    enum PaintReason {NewData, UpdateCaret,Refresh};
};

#define VERSION "0.4.99"

#include <QTextCodec>
#include <QString>
//code convert
#define G2U(s) ( QTextCodec::codecForName("GBK")->toUnicode(s) )
#define U2G(s) ( QTextCodec::codecForName("GBK")->fromUnicode(s) )
#define B2U(s) ( QTextCodec::codecForName("Big5")->toUnicode(s) )
#define U2B(s) ( QTextCodec::codecForName("Big5")->fromUnicode(s) )

extern int finPython();
extern int iniPython();
extern int iniSettings();
extern void runProgram(const QString& );
extern int iniWorkingDir();

#ifdef HAVE_PYTHON
extern PyThreadState* mainThreadState;
#endif
extern QString pathSys;
extern QString pathUsr;

// every character has a 16-bit description
// low 	 8 for colors
// hight 8 for attributes
#define COLORMASK 0x00ff
#define ATTRMASK  0xff00

// set
#define SETCOLOR(a) ( (a) & COLORMASK )
#define SETATTR(a) ( ( (a) << 8 ) & ATTRMASK )
// get
#define GETCOLOR(a) ( (a) & COLORMASK )
#define GETATTR(a) ( ( (a) & ATTRMASK ) >> 8 )

// color 8-bit
// x x x x x x x x
//   ````` * ^^^^^
//     |   |  |______foreground // 0 black,  1 red,  2 green,  3 yellow
//     |   |                    // 4 blue,  5 magenta,  6 cyan,  7 white
//     |   |___highlight 0 no, 1 highlight
//     |
//     |_____background // 0 black,  1 red,  2 green,  3 yellow
//                      // 4 blue,  5 magenta,  6 cyan,  7 white
  
#define FGMASK 0x0f
#define BGMASK 0xf0
#define HIGHLIGHTMASK 0x08

//set color
#define SETFG(c) 	( (c) & FGMASK )
#define SETBG(c) 	( ( (c) << 4 ) & BGMASK )
#define SETHIGHLIGHT(c) ( (c) | HIGHLIGHTMASK );
//get color
#define GETFG(c) 	( (c) & FGMASK )
#define GETBG(c) 	( ( (c) & BGMASK ) >> 4 )
#define GETHIGHLIGHT(c) ( (c) & HIGHLIGHTMASK )

//reverse bg and fg color
#define REVERSECOLOR(c) ( SETBG( GETFG(c) ) | SETFG( GETBG(c) ) | GETHIGHLIGHT(c) )

// mask for attr
#define BOLDMASK	0x01
#define DIMMASK		0x02
#define UNDERLINEMASK	0x08
#define BLINKMASK	0x10
#define RAPIDBLINKMASK 	0x20
#define REVERSEMASK	0x40
#define INVISIBLEMASK	0x80

//set attributes
#define SETBOLD(a) 	( (a) | BOLDMASK )
#define SETDIM(a) 	( (a) | DIMMASK )
#define SETUNDERLINE(a) ( (a) | UNDERLINEMASK )
#define SETBLINK(a) 	( (a) | BLINKMASK )
#define SETRAPIDBLINK(a) ( (a) | RAPIDBLINKMASK )
#define SETREVERSE(a) 	( (a) | REVERSEMASK )
#define SETINVISIBLE(a) ( (a) | INVISIBLEMASK )

//get attributes
#define GETBOLD(a) 	( (a) & BOLDMASK )
#define GETDIM(a) 	( (a) & BLINKMASK )
#define GETUNDERLINE(a) ( (a) & UNDERLINEMASK )
#define GETBLINK(a) 	( (a) & BLINKMASK )
#define GETRAPIDBLINK(a) ( (a) & RAPIDBLINKMASK )
#define GETREVERSE(a) 	( (a) & REVERSEMASK )
#define GETINVISIBLE(a) ( (a) & INVISIBLEMASK )

//default color 
#define NO_COLOR ( SETFG(7) | SETBG(0) )
//default attribute
#define NO_ATTR 0x04

// other definations
#ifndef NULL
#define NULL 0
#endif

#define DAE_FINISH  10001
#define DAE_TIMEOUT 10002

#define PYE_ERROR   10003
#define PYE_FINISH  10004

// some keys
#define CHAR_CR 	0x0d	// ^M
#define CHAR_LF		0x0a	// ^J
#define CHAR_FF 	0x0c	// ^L
#define CHAR_TAB 	0x09	// ^I
#define CHAR_BS		0x08	// ^H
#define CHAR_BELL	0x07	// ^G
#define CHAR_ESC	0x1b	//

#define CHAR_NORMAL	-1

// telnet state
#define TSRESOLVING		30
#define TSHOSTFOUND		31
#define TSHOSTNOTFOUND	32
#define TSCONNECTING	33
#define TSHOSTCONNECTED	34
#define TSPROXYCONNECTED	35
#define TSPROXYAUTH		36
#define TSPROXYFAIL		38
#define TSREFUSED		39
#define TSREADERROR		40
#define TSCLOSED		41
#define TSCLOSEFINISH	42
#define TSCONNECTVIAPROXY	43
#define TSEGETHOSTBYNAME	44
#define TSEINIWINSOCK	45
#define TSERROR			46
#define TSPROXYERROR	47
#define	TSWRITED		48

// proxy type
#define NOPROXY			0
#define WINGATE			1
#define SOCKS4			2
#define SOCKS5			3
#define HTTP			4

#endif	//QTERM_H

